File: app/src/main/java/org/odddev/fantlab/core/storage/StorageManager.java
package org.odddev.fantlab.core.storage;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * @author kenrube
 * @since 29.09.16
 */

public class StorageManager {

    private static final String COOKIE_KEY = "COOKIE";
    private static final String USERNAME_KEY = "USERNAME";

    private SharedPreferences sharedPreferences;

    public StorageManager(Context context) {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public void saveCookie(String cookie) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(COOKIE_KEY, cookie);
        editor.apply();
    }

    public String loadCookie() {
        return sharedPreferences.getString(COOKIE_KEY, null);
    }

    public void clearCookie() {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.remove(COOKIE_KEY);
        editor.apply();
    }

    // todo 6.32
    public void saveUsername(String username) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(USERNAME_KEY, username);
        editor.apply();
    }

    public String loadUsername() {
        return sharedPreferences.getString(USERNAME_KEY, null);
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/core/storage/StorageManager.kt
package org.odddev.fantlab.core.storage

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class StorageManager(context: Context) {

	private val COOKIE_KEY = "COOKIE"
	private val USERNAME_KEY = "USERNAME"
	private val ANONYMOUS_KEY = "ANONYMOUS"

	private val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

	fun saveCookie(cookie: String) {
		sharedPreferences.edit()
				.putString(COOKIE_KEY, cookie)
				.apply()
	}

	fun loadCookie(): String? = sharedPreferences.getString(COOKIE_KEY, null)

	fun clearCookie() {
		sharedPreferences.edit()
				.remove(COOKIE_KEY)
				.apply()
	}

	fun saveUsername(username: String) {
		sharedPreferences.edit()
				.putString(USERNAME_KEY, username)
				.apply()
	}

	fun loadUsername(): String? = sharedPreferences.getString(USERNAME_KEY, "Гость")

	fun clearUserName() {
		sharedPreferences.edit()
				.remove(USERNAME_KEY)
				.apply()
	}

	// todo разрулить схему с логином/разлогином получше
	fun saveAnonymus() {
		sharedPreferences.edit()
				.putBoolean(ANONYMOUS_KEY, true)
				.apply()
	}

	fun loadAnonymous(): Boolean = sharedPreferences.getBoolean(ANONYMOUS_KEY, false)

	fun clearAnonymous() {
		sharedPreferences.edit()
				.remove(ANONYMOUS_KEY)
				.apply()
	}
}
